### DISCUSSION TITLE
Docker with keyring on Synology

### DISCUSSION THREAD

**Discussion by papalozarou** [icloud-photos-downloader/icloud_photos_downloader#705] [Q&A]
Have been playing around with this and can get a sync going with the simple Docker command that passes both user name and password. However I'd obviously like to use this without passing the password every time. I also need to sync more than one users library, and want to run it every night.

What I can't figure out is how to use the keyring with Docker. In the docs it talks about using the `icloud` command to add to the keyring, however when I try to run this I get an error saying `--username` isn't recognised. When I use `auth add username@icloud.com` I'm not prompted for a password or for confirmation of adding to keychain. 

Browsing both the issues and discussions, I've seen a few people have raised issues, discussed having 'permanent' docker containers, or have talked about still being asked for passwords after adding to keychain. However I couldn't find any detailed instructions on how to set this up.

So a few possibly dumb questions I have:

1. Should I be setting this up as a Docker container that is always on? Or can I stop and start it?
2. Do I need to mount a volume for the keyring to be stored in? If so where is it stored within the container?
3. Can I store multiple credentials in one container? Or do I need a container for each username?

**Comment by stanthewizzard** [comment]
Same question

**Comment by AndreyNikiforov** [comment]
> 1. Should I be setting this up as a Docker container that is always on? Or can I stop and start it?

Yes, you can run docker container as service with `--watch-with-interval` parameter.

> 2. Do I need to mount a volume for the keyring to be stored in? If so where is it stored within the container?

I don't know how keyring and docker work. Pls upidate the doc once you figure it out.

> 3. Can I store multiple credentials in one container? Or do I need a container for each username?

keyring stores credentials keyed by appleid, so one keyring can store creds for multiple apple_ids. `icloudpd` currently supports only one account at a time, so you need to run multiple containers to support multiple accounts.
